commit f51ff9937bc6732ed5fc08088fdbe89ab8ed27c3
Author: Robin H. Johnson <robbat2@orbis-terrarum.net>
Date:   Sat Sep 13 16:55:01 2008 -0700

    ALSA: HDA: patch_analog: Quirk for Asus P5Q Premium/Pro boards.
    
    Use 6STACK_DIG for the AD2000BX variant of the AD1989B chip used by Asus
    on their Asus P5Q Premium and Pro boards.
    
    Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>
    Signed-off-by: Takashi Iwai <tiwai@suse.de>
    Signed-off-by: Jaroslav Kysela <perex@perex.cz>

Adjusted to apply to Debian's 2.6.26 by dann frazier <dannf@debian.org>

diff -urpN linux-source-2.6.26.orig/sound/pci/hda/patch_analog.c linux-source-2.6.26/sound/pci/hda/patch_analog.c
--- linux-source-2.6.26.orig/sound/pci/hda/patch_analog.c	2009-02-08 23:54:28.000000000 -0700
+++ linux-source-2.6.26/sound/pci/hda/patch_analog.c	2009-02-08 23:55:46.000000000 -0700
@@ -2874,6 +2874,7 @@ static struct snd_pci_quirk ad1988_cfg_t
 	SND_PCI_QUIRK(0x1043, 0x81ec, "Asus P5B-DLX", AD1988_6STACK_DIG),
 	SND_PCI_QUIRK(0x1043, 0x81f6, "Asus M2N-SLI", AD1988_6STACK_DIG),
 	SND_PCI_QUIRK(0x1043, 0x8277, "Asus P5K-E/WIFI-AP", AD1988_6STACK_DIG),
+	SND_PCI_QUIRK(0x1043, 0x8311, "Asus P5Q-Premium/Pro", AD1988_6STACK_DIG),
 	{}
 };
 
